src/main/kotlin/com/cjbooms/fabrikt/generators/controller/KtorControllerInterfaceGenerator.kt

```diff
@@ -173,17 +173,18 @@ class KtorControllerInterfaceGenerator(
 
         val addAuth = securityOption.allowsAuthenticated && options.contains(ControllerCodeGenOptionType.AUTHENTICATION)
         if (addAuth) {
-            // Using the key from the first security requirement as the auth name
-            val authName = if (operation.hasSecurityRequirements()) {
-                operation.securityRequirements.first().requirements.keys.first()
+            val authNames = if (operation.hasSecurityRequirements()) {
```

Comment:
I think this reads slightly better, but it is minor detail:

```kotlin
val authNames = if (operation.hasSecurityRequirements()) {
    operation.securityRequirements
        .filter { it.requirements.isNotEmpty() }
        .map { it.requirements.keys.first() }
} else {
    // Fall back to the global security requirements
    listOf(this.api.openApi3.securityRequirements.first().requirements.keys.first())
}.joinToString(", ") { "\"$it\"" }
```

Reply:
if you switch it to a suggestion, I can commit it